tools/automation/building_blocks/device.py

```diff
@@ -0,0 +1,90 @@
+"""
+  Contains classes which are used to store data for a device.
+
+  Ready to use, or extend subclasses as required to fill cases where more 
+  information needs to be stored.
+
+  Call to_dictionary() to retrieve all the data.
+"""
+
+import abc
+
+class Device:
+  
+  class Connection:

```

Comment:
Connection, Translation, Link classes seem like they could just be tuples? if these classes are meant to be extended, maybe define one class and inherit to be a little more DRY?